File: RobotSimulatorC#/RobotSimulatorTestApp/DataBlock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RobotSimulatorTestApp
{
    public enum Direction { N, E, S, W }

    public class DataBlock
    {
        public int Rows { get; }
        public int Columns { get; }
        public int CurrentRow { get; set; }
        public int CurrentColumn { get; set; }
        public Direction CurrentDirection { get; set; }
        public string Commands { get; }

        public DataBlock(int rows, int columns, int currentRow, int currentColumn, Direction direction, string commands)
        {
            this.Rows = rows;
            this.Columns = columns;
            this.CurrentRow = currentRow;
            this.CurrentColumn = currentColumn;
            this.CurrentDirection = direction;  
            this.Commands = commands;
        }
    }
}

File: 1_RobotSimulatorJava/DataBlock.java
import java.util.ArrayList;
import java.util.List;

enum Direction {
    N, E, S, W;
}

class DataBlock {
    private int rows;
    private int columns;
    private int currentRow;
    private int currentColumn;
    private Direction currentDirection;
    private String commands;

    // Getter for rows
    public int getRows() {
        return rows;
    }

    // Getter for columns
    public int getColumns() {
        return columns;
    }

    // Getter for currentRow
    public int getCurrentRow() {
        return currentRow;
    }

    // Setter for currentRow
    public void setCurrentRow(int currentRow) {
        this.currentRow = currentRow;
    }

    // Getter for currentColumn
    public int getCurrentColumn() {
        return currentColumn;
    }

    // Setter for currentColumn
    public void setCurrentColumn(int currentColumn) {
        this.currentColumn = currentColumn;
    }

    // Getter for currentDirection
    public Direction getCurrentDirection() {
        return currentDirection;
    }

    // Setter for currentDirection
    public void setCurrentDirection(Direction currentDirection) {
        this.currentDirection = currentDirection;
    }

    // Getter for commands
    public String getCommands() {
        return commands;
    }

    // Setter for commands
    public void setCommands(String commands) {
        this.commands = commands;
    }


    public DataBlock(int rows, int columns, int currentRow, int currentColumn, Direction currentDirection, String commands) {
        this.rows = rows;
        this.columns = columns;
        this.currentRow = currentRow;
        this.currentColumn = currentColumn;
        this.currentDirection = currentDirection;
        this.commands = commands;
    }

}

